features/admin.template-core.v1/components/template-dynamic-form.tsx

```diff
@@ -0,0 +1,251 @@
+import has from "lodash-es/has";
+import pick from "lodash-es/pick";
+import set from "lodash-es/set";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { Grid } from "semantic-ui-react";
+import { FormDynamicField } from "./form-dynamic-field";
+import useInitializeHandlers, { CustomInitializeFunction } from "./forms/handlers/initialize/use-initialize-handlers";
+import useSubmissionHandlers, { CustomSubmissionFunction } from "./forms/handlers/submission/use-submission-handlers";
+import useValidationHandlers, { CustomValidationsFunction } from "./forms/handlers/validation/use-validation-handlers";
+import { DynamicFieldInterface, DynamicFormInterface } from "../models/dynamic-fields";
+
+/**
+ * Prop types of the `TemplateDynamicForm` component.
+ */
+export interface TemplateDynamicFormPropsInterface extends IdentifiableComponentInterface {
+    /**
+     * Custom validation functions for form validations.
+     */
+    customValidations?: CustomValidationsFunction;
+    /**
+     * Custom initialize functions.
+     */
+    customInitializers?: CustomInitializeFunction;
+    /**
+     * Custom submission handler functions.
+     */
+    customSubmissionHandlers?: CustomSubmissionFunction;
+    /**
+     * Definition of the dynamic form.
+     */
+    form: DynamicFormInterface;
+    /**
+     * Initial values for the form fields.
+     */
+    initialFormValues: Record<string, unknown>;
+    /**
+     * Template payload values.
+     */
+    templatePayload: Record<string, unknown>;
+    /**
+     * i18n key of the form main button text.
+     */
+    buttonText: string;
+    /**
+     * Function to handle form submission.
+     */
+    onFormSubmit: (values: Record<string, unknown>, callback: () => void) => void;
+    /**
+     * Loading status for the wizard.
+     */
+    isLoading: boolean;
+    /**
+     * Prop to indicate whether the form is read-only.
+     */
+    readOnly: boolean;
+}
+
+/**
+ * Template Dynamic form component.
+ *
+ * @param Props - Props to be injected into the component.
+ */
+export const TemplateDynamicForm: FunctionComponent<TemplateDynamicFormPropsInterface> = ({
+    customInitializers,
+    customSubmissionHandlers,
+    customValidations,
+    form,
+    initialFormValues,
+    templatePayload,
+    buttonText,
+    onFormSubmit,
+    isLoading,
+    readOnly,
+    ["data-componentid"]: componentId = "template-dynamic-form"
+}: TemplateDynamicFormPropsInterface): ReactElement => {
+
+    const { validate } = useValidationHandlers(customValidations);
+    const { initialize } = useInitializeHandlers(customInitializers);
+    const { submission } = useSubmissionHandlers(customSubmissionHandlers);
+
+    const { UIConfig } = useUIConfig();
+    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+    const [ isSubmitting, setIsSubmitting ] = useState(false);
```

Comment:
Add the state type

Reply:
https://github.com/wso2/identity-apps/pull/5896/commits/bd0cc60f01d23d858ec1958ff366608ba7732698